# Exercise-2
Add the specified code for each code cell, running the cells _in order_.

Open the included **`data/words_small.txt`** file (containing a "small" list of English words) and `read()` it into a single string variable **`words_str`**. Then print the first 100 _characters_ of the `words_str` variable (use a slice operator!)

In [3]:
ws=open('data/words_small.txt')
words_str=ws.read()
print(words_str[:100])

abbreviate
abbreviation
abdomen
able
aboard
abroad
absent
absolute
abuse
academic
academy
accelerate


**Using a loop**, count the number of times the newline character `\n` appears in the `words_str`. This should tell you the number of words in the file (plus or minus one). You can check your work with the built-in `count()` string method (but use the loop for practice!)

In [15]:
n=0
for i in words_str:
    if '\n' in i:
        n+=1
print(n)
print(words_str.count('\n'))

1520
1520


Create a variable **`letter_counts`** that is a dictionary whose keys are characters (letters) and whosevalues are the number of times that character appears in `words_str`. Print the dictionary after you make it.
- To do this, start with an empty dictionary, then loop through the letters one at a time. At each letter, increase the value associated with that letter key by one.
- Watch out for if the letter is not in the dictionary yet!
- Also _delete_ the entry for the newline `\n` character

In [17]:
letter_counts={}
strin='abcdefghijklmnopqrstuvwxyz'
for i in strin:
    letter_counts[i]=words_str.count(i)
letter_counts

{'a': 703,
 'b': 188,
 'c': 328,
 'd': 314,
 'e': 1118,
 'f': 155,
 'g': 213,
 'h': 295,
 'i': 515,
 'j': 16,
 'k': 126,
 'l': 457,
 'm': 223,
 'n': 550,
 'o': 598,
 'p': 224,
 'q': 9,
 'r': 642,
 's': 504,
 't': 604,
 'u': 279,
 'v': 85,
 'w': 161,
 'x': 23,
 'y': 189,
 'z': 15}

Print out all the count in the `letter_counts` dictionary in the format `"A appears 703 times"`, each on a new line, in alphabetical order. _Hint_: iterate through the sorted keys!

In [20]:
for i in letter_counts:
    print(i," appears ",letter_counts[i]," times")

a  appears  703  times
b  appears  188  times
c  appears  328  times
d  appears  314  times
e  appears  1118  times
f  appears  155  times
g  appears  213  times
h  appears  295  times
i  appears  515  times
j  appears  16  times
k  appears  126  times
l  appears  457  times
m  appears  223  times
n  appears  550  times
o  appears  598  times
p  appears  224  times
q  appears  9  times
r  appears  642  times
s  appears  504  times
t  appears  604  times
u  appears  279  times
v  appears  85  times
w  appears  161  times
x  appears  23  times
y  appears  189  times
z  appears  15  times


Print which letter from the `letter_counts` dictionary has a **value** of `598`. _Hint:_ use a loop to look through all the _items_ until you find the right one, then print that!

In [22]:
for i in letter_counts:
    if letter_counts[i]==598:
        print(i)

o


Create a variable **`count_to_letter_dict`** that is a dictionary whose **keys** are the counts and whose **values** are the letters that have that count. This is a _reverse lookup_ dictionary. Hint: loop through items in `letter_counts` and add an item to your new dictionary for each element. You can assume no duplicated counts.

In [36]:
count_to_letter_dict = {}
for letter,count in letter_counts.items():
    count_to_letter_dict[count] = letter  # assumes no duplicates, otherwise would need lists

Print the **10** most common letters and their counts with the format `"A appears 703 times"`. _Hint:_ you'll need to sort the keys to `count_to_letter_dict` and take the first 10 of those.

In [34]:
sorted_keys = sorted(count_to_letter_dict.keys(), reverse=True) 
for count in sorted_keys[:10]:  # slice!
    print(count_to_letter_dict[count].upper(), "appears", count, "times")

E appears 1118 times
A appears 703 times
R appears 642 times
T appears 604 times
O appears 598 times
N appears 550 times
I appears 515 times
S appears 504 times
L appears 457 times
C appears 328 times


Bonus: repeat your analysis on the `data/words_huge.txt` file! Are the top letters the same?